### PR TITLE
chore: inject schema using schema id for cas

### PR DESCRIPTION
**Pull request by lihaosky**
### Description 
* Inject schema for cas statement if schema id is provided.
* Update QTT test to use `SchemaRegisterInjector` as well.
* Disabled a few failing QTT tests as it also fails in current master branch (https://github.com/confluentinc/ksql/issues/8465)

### Testing done 
* Unit test
* QTT test

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java [hunk 1]
```diff
@@ -74,67 +74,73 @@ public SchemaResult getKeySchema(
...
         id = schemaId.get();
       } else {
+        subject = getSRSubject(topicName.get(), isKey);
         id = srClient.getLatestSchemaMetadata(subject).getId();
       }
 
       final ParsedSchema schema = srClient.getSchemaBySubjectAndId(subject, id);
-      return fromParsedSchema(topicName, id, schema, expectedFormat, serdeFeatures, isKey,
-          schemaId.isPresent());
+      return fromParsedSchema(topicName, id, schemaId, schema, expectedFormat, serdeFeatures,
```
Comment (spena) [inline ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java:109]: `id` and `schemaId` are the same. Can this become into one parameter?

Comment (lihaosky) [reply ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java:109]: Will take a look how to refactor

Comment (lihaosky) [reply ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegistryTopicSchemaSupplier.java:109]: Actually `schemaId` is optional and is used to indicate whether schemaId is provided by user to fetched from latest schema. Maybe I can refactor to call it `isIdProvided`

File: ksqldb-serde/src/test/java/io/confluent/ksql/serde/protobuf/KsqlProtobufSerializerTest.java [hunk 2]
```diff
@@ -160,6 +201,44 @@ public void shouldSerializeBytesField() {
...
+    // When:
+    final Exception e = assertThrows(
+        SerializationException.class,
+        () -> serializer.serialize(SOME_TOPIC, RANDOM_NAME_STRUCT_RECORD)
+    );
+
+    // Then:
+    assertThat(e.getMessage(),
+        containsString("Error serializing message to topic: bob. "
+            + "Failed to serialize Protobuf data from topic bob"));
```
[the remark this answers is not in the document]
Comment (lihaosky) [reply ksqldb-serde/src/test/java/io/confluent/ksql/serde/protobuf/KsqlProtobufSerializerTest.java:240]: This is actually dependent PR's code and removed. Let me rebase this PR...

File: ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java [hunk 3]
```diff
@@ -347,14 +347,13 @@ private void registerSchema(
...
-          PersistenceSchema.from(schema, serdeFeatures)
-      );
-
       try {
-        SchemaRegistryUtil.registerSchema(srClient, parsedSchema, topic, subject, isKey);
+        final SchemaTranslator translator = format.getSchemaTranslator(formatInfo.getProperties());
+        final ParsedSchema parsedSchema = translator.toParsedSchema(
+            PersistenceSchema.from(schema, serdeFeatures)
+        );
+        int id = SchemaRegistryUtil.registerSchema(srClient, parsedSchema, topic, subject, isKey);
```
Comment (spena) [inline ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java:355]: what do you do with the `id`?

File: ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java [hunk 4]
```diff
@@ -282,14 +286,16 @@ private static void initializeTopics(
 
       topic.getKeySchema().ifPresent(schema -> {
         try {
-          srClient.register(KsqlConstants.getSRSubject(topic.getName(), true), schema);
+          int id = srClient.register(KsqlConstants.getSRSubject(topic.getName(), true), schema);
+          System.out.print("h");
```
Comment (spena) [inline ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java:290]: where is the `id` use? What about the print call?

File: ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/QueryTranslationTest.java [hunk 5]
```diff
@@ -83,11 +83,15 @@
    */
   @SuppressWarnings("unused")
   public QueryTranslationTest(final String name, final TestCase testCase) {
+
     this.testCase = requireNonNull(testCase, "testCase");
   }
 
   @Test
   public void shouldBuildAndExecuteQueries() {
+    if (!testCase.getName().contains("DECIMAL - key - inference - default precision")) {
```
Comment (spena) [inline ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/QueryTranslationTest.java:92]: Is this usd for debugging?

File: ksqldb-functional-tests/src/test/resources/query-validation-tests/kafka.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
         "CREATE STREAM INPUT (K STRING KEY, foo INT) WITH (WRAP_SINGLE_VALUE=true, kafka_topic='input', value_format='KAFKA');"
       ],
       "expectedException": {
-        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "type": "io.confluent.ksql.util.KsqlException",
```
Comment (spena) [inline ksqldb-functional-tests/src/test/resources/query-validation-tests/kafka.json:10]: Can the `KsqlStatementException` stay? There is a check at the end of the `KsqlResource.handleKsqlStatements()` method that returns a different value depending on the exception.

This question applies for the rest of the tests that changes to this new exception.

Comment (lihaosky) [reply ksqldb-functional-tests/src/test/resources/query-validation-tests/kafka.json:10]: Sure. Will catch `KsqlException` and rethrow

File: ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/QueryTranslationTest.java (hunk 5 above)
Comment (lihaosky) [reply ksqldb-functional-tests/src/test/java/io/confluent/ksql/test/QueryTranslationTest.java:92]: Yeah. Will remove

File: ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java (hunk 4 above)
Comment (lihaosky) [reply ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java:290]: Will remove

File: ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java (hunk 3 above)
Comment (lihaosky) [reply ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java:355]: Will remove

File: ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java [hunk 7]
```diff
@@ -347,14 +347,13 @@ private void registerSchema(
...
+        System.out.print("h");
```
[the remark this answers is not in the document]
Comment (lihaosky) [reply ksqldb-engine/src/main/java/io/confluent/ksql/schema/ksql/inference/SchemaRegisterInjector.java:356]: Will remove

**Review by spena — APPROVED** [review]
looks good

**Pull request merged**